eth/handler.go

```diff
@@ -452,6 +468,18 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		// Filter out any explicitly requested headers, deliver the rest to the downloader
 		filter := len(headers) == 1
 		if filter {
+			// Check for any responses not matching our whitelist
+			if whitelist := pm.whitelist; whitelist != nil {
+				if expected, ok := (*whitelist)[headers[0].Number.Uint64()]; ok {
+					actual := headers[0].Hash()
+					if !bytes.Equal(expected.Bytes(), actual.Bytes()) {
+						p.Log().Debug("dropping peer with non-matching whitelist block", "number", headers[0].Number.Uint64(), "expected", expected, "actual", actual)
```

Comment:
We shouldn't see this very often, and if we're actually in the crisis-mode where this is enabled, I think it would be nice to get the notification, so I'd suggest `Info` instead of `Debug`

```suggestion
						p.Info().Debug("dropping peer with non-matching whitelist block", "number", headers[0].Number.Uint64(), "expected", expected, "actual", actual)
```